Entities.js

```diff
@@ -1,32 +1,62 @@
 import { StyleSheet, Text, View, Image } from 'react-native';
 import { Dimensions, Keyboard } from 'react-native';
-import { React, Component, useEffect, useState, useRef } from 'react';
+import { React, Component, useEffect, useState, useRef, PureComponent } from 'react';
 import { UseOnKeyPress, UseOnKeyRelease } from "./Key.js"
-import { CreateCircle, CreateBrick, DeleteEntity } from "./Dodge.js"
-
+import { rectangleCollision, circleCollision, Intersection } from './Intersections.js';
+import DodgeBall from "./Dodge.js"
+import Matter from "matter-js"
+import generateUniqueId from 'generate-unique-id';
 
 const MAXwidth = Dimensions.get('window').width;
 const MAXheight = Dimensions.get('window').height;
-let counter = 0
-let GlobalX = 400
-let GlobalY = 400
 let playerHeight = 150
 let playerWidth = 150
-let input = [0,0,0,0]
+let input = [0, 0, 0, 0]
 //      left,right,up,down (w,a,s,d) each stores a value, 0 = not pressed
 
+// TODO: fix intersections between circle and rectangle
 //count intersections with players
-const Counter = () => {
-  console.log({counter})
-} 
+class Shape extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.height = 75;
+    this.width = 75;
+    this.backgroundColor = "red"
+    let randomNum = Math.random() * 3;
+    this.momentum = [randomNum, 3 - randomNum]
+  }
+  update() {
+    // all shapes move and have momentum
+    this.x += this.momentum[0];
+    this.y += this.momentum[1];
+  }
+}
+
+function CreateSprite(props,) {
+  return (
+    <Image
+      source={props.image}
+      style={{
+        position: 'absolute',
+        borderRadius: props.borderRadius,
+        width: props.width,
+        height: props.height,
+        left: props.x,
+        top: props.y,
+        backgroundColor: props.backgroundColor,
+      }}
+    />
+  );
+}
+
 
 //function to detect wasd input on computer
-function MoveKnight (){
+function MoveKnight() {
```

Comment:
This function is more like "SetupKeyListeners". And probably doesn't need to be rendered as a component (in Dodge.js). You can probably just call it from within the constructor for Knight.

Reply:
This was part of the whole workaround for the rules of hooks, which I still don't know very well